source/_patterns/03-templates/body/body.twig

```diff
@@ -0,0 +1,36 @@
+{%- set attributes = attributes|default({})|merge({class: attributes.class|default([]) }) -%}
+
+<body {%- include('atoms-attributes') -%}>
+
+    {# site header goes here#}
```

Comment:
Thinking it should be a grid of slots, like:

```
┌─────────┐
│ Header  │
├─────────┤
│   Top   │
├─────────┤
│ Content │
├─────────┤
│ Bottom  │
├─────────┤
│ Footer  │
└─────────┘
```

(Top/Bottom have padding in contrast to Header/Footer.)

And later:

```
┌──────────────┐
│    Header    │
├──────────────┤
│     Top      │
├──────┬───────┤
│ Left │ Right │
├──────┴───────┤
│    Bottom    │
├──────────────┤
│    Footer    │
└──────────────┘
```

(Same applies to it if the grid doesn't cover the whole page.)

Reply:
I envisage the top level grid will encompass the whole page. Although we don't have [subgrids](https://www.w3.org/TR/css-grid-2/#subgrids) yet, this should in theory allow them to be defined anywhere on the page once they're supported, without a lot of reengineering of the top level. Conversely, if non-grid behaviour is desired for a section of the page, that grid section can be set to the equivalent of 100% inline size and things can be treated as required within that.

Taking this approach, the top level grid would be a single column grid, that could have other grids embedded as necessary. For example, in your 2-column example above, the content grid cell would include a 2-column grid.

Reply:
Cool, so sub-grids through templates rather than CSS. That means we don't need the top-level grid, as it's just a single block (implementations continue to be like https://github.com/libero/browser/blob/dc4c0b98ec6b722821a9e1f8532908ee2a8c2323/templates/page.html.twig#L21).

Reply:
Probably want to remove the wrapper from the current ContentHeader, it should be wrapped in a wrapper instead (it's a 'full-width' single column).

Reply:
CSS will be needed to create a proper subgrid. The way I'm thinking of it, the grid included via a template could have the option of being specified as a subgrid, which would support configuring CSS to enable the subgrid appropriately. It could get complicated, and until it lands in browsers we can't be sure, but this general approach seems future friendly to me without too much up front work. One upshot is I think we do need the top level grid.

Reply:
Having discussed this with @thewilkybarkid we'll keep this as simple as possible for the first iteration: the `<body>` template will be supplied by https://github.com/libero/browser and the single-column grid template by this pattern library.

Reply:
Having been thinking about and discussing grids for the last couple of days, I think the grid for the production dashboard will differ from the grid for the reading experience. It may be that they share some common aspects which it might make sense to abstract, we'll have to see.